gbpservice/nfp/configurator/drivers/loadbalancer/v2/haproxy/haproxy_driver.py

```diff
@@ -0,0 +1,592 @@
+                if not member.pool:
+                    member.pool = ret
+                members.append(member)
+            args.update({
+                'members': members
+            })
+        if pool_dict.get('healthmonitor'):
+            healthmonitor = self.get_healthmonitor_octavia_model(
+                pool_dict.get('healthmonitor'))
+            if not healthmonitor.pool:
+                healthmonitor.pool = ret
+            args.update({
+                'health_monitor': healthmonitor
+            })
+
+        # TODO: L7Policy are not added
+        args.update({
+            'protocol': pool.protocol,
+            'lb_algorithm': pool.lb_algorithm,
+            'session_persistence': pool.session_persistence,
+        })
+        ret = self._update(ret, args)
+        return ret
+
+    # Translate member neutron model dict to octavia model
+    def get_member_octavia_model(self, member_dict):
+        member = n_data_models.Member.from_dict(
+            copy.deepcopy(member_dict)
+        )
+        ret = o_data_models.Member()
+        args = {
+            'id': member.id,
+            'project_id': member.tenant_id,
+            'pool_id': member.pool_id,
+            'ip_address': member.address,
+            'protocol_port': member.protocol_port,
+            'weight': member.weight,
+            'enabled': member.admin_state_up,
+            'subnet_id': member.subnet_id,
+            'operating_status': member.operating_status,
+        }
+        if member_dict.get('pool'):
+            pool = self.get_pool_octavia_model(member_dict.get('pool'))
+            args.update({
+                'pool': pool
+            })
+        ret = self._update(ret, args)
+        return ret
+
+    # Translate HealthMonitor neutron model dict to octavia model
+    def get_healthmonitor_octavia_model(self, hm_dict):
+        hm = n_data_models.HealthMonitor.from_dict(
+            copy.deepcopy(hm_dict)
+        )
+        ret = o_data_models.HealthMonitor()
+        args = {
+            'id': hm.id,
+            'project_id': hm.tenant_id,
+            'type': hm.type,
+            'delay': hm.delay,
+            'timeout': hm.timeout,
+            'rise_threshold': hm.max_retries,
+            'fall_threshold': hm.max_retries,
+            'http_method': hm.http_method,
+            'url_path': hm.url_path,
+            'expected_codes': hm.expected_codes,
+            'enabled': hm.admin_state_up
+        }
+        if hm_dict.get('pool'):
+            pool = self.get_pool_octavia_model(hm_dict.get('pool'))
+            args.update({
+                'pool': pool,
+                'pool_id': pool.id
+            })
+        ret = self._update(ret, args)
+        return ret
+
+
+class HaproxyLoadBalancerDriver(n_driver_base.LoadBalancerBaseDriver,
+                                LbGenericConfigDriver,
+                                base_driver.BaseDriver):
+    service_type = 'loadbalancerv2'
```

Comment:
Following GBP files are changed to support new service type. 
- gbpservice/neutron/services/servicechain/plugins/msc/drivers/simplechain_driver.py
- gbpservice/neutron/services/servicechain/plugins/ncp/node_drivers/heat_node_driver.py

@songole - Since this is a change in GBP code, we need to propose these changes also right?